packages/odyssey-react-mui/src/ui-shell/UiShell.tsx

```diff
@@ -0,0 +1,159 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {
+  memo,
+  useEffect,
+  useState,
+  type ReactNode,
+  type SetStateAction,
+} from "react";
+import { ErrorBoundary, ErrorBoundaryProps } from "react-error-boundary";
+
+import { SideNav, type SideNavProps } from "../labs/SideNav";
+import { TopNav, type TopNavProps } from "../labs/TopNav";
+import { OdysseyProvider } from "../OdysseyProvider";
+import { ShadowDomElements } from "../web-component/shadow-dom";
+
+const appContainerStyles = {
+  flexGrow: "1",
+};
+
+const uiShellContainerStyles = {
+  display: "flex",
+};
+
+export type UiShellComponentProps = {
+  sideNavProps?: Omit<SideNavProps, "logo" | "footerComponent">;
+  topNavProps: Omit<
+    TopNavProps,
+    "AdditionalNavItemComponent" | "SearchFieldComponent"
+  >;
+};
+
+export const defaultComponentProps: UiShellComponentProps = {
+  topNavProps: {
+    topNavLinkItems: [],
+  },
+} as const;
+
+export type UiShellProps = {
+  /**
+   * React app component that renders as children in the correct location of the shell.
+   */
+  appComponent: ReactNode;
+  /**
+   * Notifies when a React rendering error occurs. This could be useful for logging, flagging "p0"s, and recovering UI Shell when errors occur.
+   */
+  onError?: ErrorBoundaryProps["onError"];
+  /**
+   * Notifies when subscribed to prop changes.
+   *
+   * UI Shell listens to prop updates, and it won't subscribe synchronously. Because of that, this callback notifies when that subscription is ready.
+   */
+  onSubscriptionCreated: () => void;
+  /**
+   * Components that will render as children of various other components such as the top nav or side nav.
+   */
+  optionalComponents?: {
+    additionalTopNavItems?: TopNavProps["AdditionalNavItemComponent"];
+    footer?: SideNavProps["footerComponent"];
+    logo?: SideNavProps["logo"];
+    searchField?: TopNavProps["SearchFieldComponent"];
+  };
+  /**
+   * This is a callback that provides a subscriber callback to listen for changes to state.
+   * It allows UI Shell to listen for state changes.
+   *
+   * The props coming in this callback go directly to a React state; therefore, it shares the same signature and provides a previous state.
+   */
+  subscribeToPropChanges: (
+    subscriber: (componentProps: SetStateAction<UiShellComponentProps>) => void,
+  ) => () => void;
+} & ShadowDomElements;
+
+/**
+ * Our new Unified Platform UI Shell.
+ *
+ * This includes the top and side navigation as well as the footer and provides a spot for your app to render into.
+ *
+ * If an error occurs, this will revert to only showing the app.
+ */
+const UiShell = ({
+  appComponent,
+  appRootElement,
```

Comment:
This is the prop I'm trying to highlight. It exists in this object, but is not defined (and documented) in `UiShellProps` above.

Reply:
That was confusing. These are from the return value of the function that creates the two elements created in `renderReactInWebComponent`.

Added docs to those return values which are inherited in the type definition!
![image](https://github.com/user-attachments/assets/10482bdd-b89c-4e8c-a268-ba52f7003b18)